torch/autograd/gradcheck.py

```diff
@@ -1033,19 +1004,16 @@ def gradcheck_helper(func, inputs, eps, atol, rtol, check_sparse_nnz, nondet_tol
     check_inputs(tupled_inputs, check_sparse_nnz)
     func_out = func(*tupled_inputs)
 
-    if fast_mode and has_complex_inputs_or_outputs(tupled_inputs, func_out):
-        raise NotImplementedError("Fast mode for gradcheck and gradgradcheck is currently only implemented"
-                                  " for R to R functions.")
+    # if fast_mode and has_complex_inputs_or_outputs(tupled_inputs, func_out):
+    #     raise NotImplementedError("Fast mode for gradcheck and gradgradcheck is currently only implemented"
+    #                               " for R to R functions.")
 
     outputs = _differentiable_outputs(func_out)
     check_outputs(outputs)
-
-    if fast_mode:
-        fast_gradcheck(func, func_out, tupled_inputs, outputs, eps, rtol,
-                       atol, check_grad_dtypes, nondet_tol)
-    else:
-        slow_gradcheck(func, func_out, tupled_inputs, outputs, eps, rtol,
-                       atol, check_grad_dtypes, nondet_tol)
+    complex_indices = [i for i, o in enumerate(outputs) if o.is_complex()]
```

Comment:
Do we actually have any example of a function that returns mixed real and complex outputs? Could we assume they are either all real or all complex?

Reply:
As discussed offline, keeping this since it is tested and doesn't add too much complexity